src/Umbraco.Web/Unversion/UnversionTask.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Web.Scheduling;
+
+namespace Umbraco.Web.Unversion
+{
+
+    public class UnversionTask : RecurringTaskBase
+    {
+        private IRuntimeState _runtime;
+        private IProfilingLogger _logger;
+        private IContentService _contentService;
+        private IUnversionService _unversionService;
+
+        public UnversionTask(IBackgroundTaskRunner<RecurringTaskBase> runner, int delayBeforeWeStart, int howOftenWeRepeat, IRuntimeState runtime, IProfilingLogger logger, IContentService contentService, IUnversionService unversionService)
+            : base(runner, delayBeforeWeStart, howOftenWeRepeat)
+        {
+            _runtime = runtime;
+            _logger = logger;
+
+            _contentService = contentService;
+            _unversionService = unversionService;
+        }
+
+        public override bool PerformRun()
+        {
+            if (_runtime.ServerRole != Core.Sync.ServerRole.Master && _runtime.ServerRole != Core.Sync.ServerRole.Single)
+            {
+                _logger.Debug<UnversionTask>("Unversion background task does not run when server role '{ServerRole}' is not Master or Single", _runtime.ServerRole);
+                return true; // We return true to try again as the server role may change!
+            }
+
+            var allContent = new List<IContent>();
+
+            // Add TraceDuration - so users can see if this is taking a long time to calculate
+            using (_logger.TraceDuration<UnversionTask>("Calculating number of content nodes in site for Unversion", "Calculated number of nodes in site for Unversion"))
+            {               
+                // Get all content nodes
+                // Worried this could be bad for performance...
```

Comment:
Make sure to test this on a large site, I have the same worry :)

Reply:
Yes I will do some testing shortly

Reply:
All of this needs to be done in bulk, this cannot happen by selecting each content item in the tree. Lets please revisit all of this and how it should be handled.